hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -3366,3 +3376,64 @@ func (r *NodePoolReconciler) reconcileMirroredConfigs(ctx context.Context, logr
 	}
 	return nil
 }
+
+// SetPerformanceProfileConditions checks for performance profile status updates, and reflects them in the nodepool status conditions
+func (r *NodePoolReconciler) SetPerformanceProfileConditions(ctx context.Context, logger logr.Logger, nodePool *hyperv1.NodePool, controlPlaneNamespace string, toDelete bool) error {
+	if toDelete {
+		ppConds := []string{
+			hyperv1.NodePoolPerformanceProfileTuningAvailableConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningProgressingConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningUpgradeableConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningDegradedConditionType,
+		}
+		for _, ppCond := range ppConds {
+			removeStatusCondition(&nodePool.Status.Conditions, ppCond)
+		}
+		return nil
+	}
+	// Get performance profile status configmap
+	cmList := &corev1.ConfigMapList{}
+	if err := r.Client.List(ctx, cmList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodeTuningGeneratedPerformanceProfileStatusLabel: "true"}),
+		Namespace:     controlPlaneNamespace,
+	}); err != nil {
+		return err
+	}
+	if len(cmList.Items) > 1 {
+		return fmt.Errorf("there cannot be more than one PerformanceProfile ConfigMap status per NodePool. found: %d NodePool: %s", len(cmList.Items), nodePool.Name)
+	}
+	if len(cmList.Items) == 0 {
+		// Only log here and do not return an error because it might take sometime for NTO to
+		// generate the ConfigMap with the PerformanceProfile status.
+		// The creation of the ConfigMap itself triggers the reconciliation loop which eventually calls
+		// this flow again.
+		logger.V(4).Info("no PerformanceProfile ConfigMap status found", "Namespace", controlPlaneNamespace, "NodePool", nodePool.Name)
+		return nil
+	}
+	performanceProfileStatusConfigMap := cmList.Items[0]
+	statusRaw, ok := performanceProfileStatusConfigMap.Data["status"]
+	if !ok {
+		return fmt.Errorf("status not found in performance profile status configmap")
+	}
+	status := &performanceprofilev2.PerformanceProfileStatus{}
+	if err := yaml.Unmarshal([]byte(statusRaw), status); err != nil {
+		return fmt.Errorf("failed to decode the performance profile status: %w", err)
+	}
+
+	for _, ppCond := range status.Conditions {
+		condition := hyperv1.NodePoolCondition{
+			Type:               fmt.Sprintf("%s/%s", hyperv1.NodePoolPerformanceProfileTuningConditionTypePrefix, ppCond.Type),
+			Status:             ppCond.Status,
+			Reason:             ppCond.Reason,
+			Message:            ppCond.Message,
+			ObservedGeneration: nodePool.Generation,
+		}
+		oldCondition := FindStatusCondition(nodePool.Status.Conditions, condition.Type)
+
+		// Will set the condition only if it was not set previously, or has changed
+		if oldCondition == nil || reflect.DeepEqual(oldCondition, condition) {
```

Comment:
We try to avoid using reflect because of the performance, for that, you can use the field `observedGeneration` which basically tells you if the status has changed from last time.